datafusion/datasource/src/file_format.rs

```diff
@@ -109,37 +108,10 @@ pub trait FileFormat: Send + Sync + fmt::Debug {
         not_impl_err!("Writer not implemented for this format")
     }
 
-    /// Check if the specified file format has support for pushing down the provided filters within
-    /// the given schemas. Added initially to support the Parquet file format's ability to do this.
-    fn supports_filters_pushdown(
```

Comment:
Binning specialized code that was also leaking parquet stuff through DataSource and into TableProvider 😄

Reply:
yes I agree 

Since [`FileFormat`](https://docs.rs/datafusion/latest/datafusion/datasource/file_format/trait.FileFormat.html) is a pub trait, this is technically a breaking API change, but I do think it was a parquet specific optimization

I recommend we mark this PR as an API change and add a note to the upgrade guide https://github.com/apache/datafusion/blob/main/docs/source/library-user-guide/upgrading.md

I think it should basically say if you implemented `FileFormat` (which probably no one did) and ListingTable you will have to implement the newly added `ExecutionPlan::try_pushdown_filter` method into your execution plan directly if you want the filters to be pushed down